chalk-engine/src/lib.rs

```diff
@@ -290,25 +290,3 @@ impl DepthFirstNumber {
         DepthFirstNumber { value }
     }
 }
-
-/// Because we recurse so deeply, we rely on stacker to
```

Comment:
We may still need stack protection, although I agree it should be far, far less. But it's possible for us to recurse quite deeply across traits.

Reply:
Where do you think it should go? `ensure_answer_recursively`?

Reply:
I've also been thinking about how to maybe even remove *that* recursion though. But that wouldn't be for this PR.